packages/xdl/src/Project.ts

```diff
@@ -1578,6 +1583,19 @@ function _logPackagerOutput(projectRoot: string, level: string, data: Object) {
   }
 }
 
+function _isIgnorableMetroConsoleOutput(output: string) {
+  // As of React Native 0.61.x, Metro prints console logs from the device to console, without
+  // passing them through the custom log rgeporter.
+  //
+  // Managed apps have a separate remote logging implementation included in the Expo SDK,
+  // (see: _handleDeviceLogs), so we can just ignore these device logs from Metro.
+  // if (/^ () /)
+  //
+  // These logs originate from:
+  // https://github.com/facebook/metro/blob/e8181fb9db7db31adf7d1ed9ab840f54449ef238/packages/metro/src/lib/logToConsole.js#L50
+  return /^\s+(INFO|WARN|LOG|GROUP|DEBUG) /.test(output);
```

Comment:
it's a bit unfortunate that we need to match specifically on these strings because if a user were to do `console.log('WARN: something')` or `console.log('INFO: blah')` it seems that we would filter out their log message

Reply:
This filtering only applies to the raw output from Metro, which normally is JSON, except in case of these new device console logs, which bypass the log reporter and are directly written to the console. https://github.com/facebook/metro/pull/465 will fix that and remove the need to match by a prefix in the future versions.

In any case, every message logged in the app itself will be printed, even if it has a prefix like “ WARN “, because the logs from the device/simulator are sent directly to Expo CLI by the SDK, and this code path doesn’t touch them at all.